lib/src/gotrue_client.dart

```diff
@@ -687,6 +687,11 @@ class GoTrueClient {
       );
       return completer.future;
     } catch (error, stack) {
+      if (error is AuthException) {
+        if (error.message == 'Invalid Refresh Token: Refresh Token Not Found') {
+          signOut();
+        }
+      }
       completer.completeError(error, stack);
       return completer.future;
```

Comment:
Thanks for this! About whether we should keep throwing an error or not, it seems like [gotrue-js just does not throw error](https://github.com/supabase/gotrue-js/blob/master/src/GoTrueClient.ts#L1099) for any kind of error. We can probably just align with this behavior and just do something like the following, but what do you think?

```suggestion
     debugPrint(error.toString());
      completer.complete(null);
      return completer.future;
```

Reply:
But your suggested change, wouldn't sign the user out. I think we should only print the error in that specific invalid refresh token case and keep the behavior for other errors.

Reply:
> But your suggested change, wouldn't sign the user out.

Sorry, forgot to include it in 😂

> I think we should only print the error in that specific invalid refresh token case and keep the behavior for other errors.

Sure, that works too!

Reply:
I just noticed `debugPrint` is a method from Flutter. What about instead emitting an event and use the existing logging mechanism in supabase-flutter to print the message? I think that's a better error handling than just printing it.

Reply:
@Vinzent03 
Sounds good!